### ISSUE TITLE
403 Client Error: Forbidden for url: https://areena.yle.fi/1-1570236

### ISSUE THREAD

**Issue by chenrui333** [aajanki/yle-dl#333]
yle-dl version: 20221015

error stacktrace for `yle-dl --showtitle https://areena.yle.fi/1-1570236`
```
Traceback (most recent call last):
  File "/userpath/bin/yle-dl", line 8, in <module>
    sys.exit(main())
  File "/userpath/lib/python3.10/site-packages/yledl/yledl.py", line 477, in main
    res = execute_action(
  File "/userpath/lib/python3.10/site-packages/yledl/yledl.py", line 287, in execute_action
    print_lines(dl.get_titles(url, io, stream_filters.latest_only))
  File "/userpath/lib/python3.10/site-packages/yledl/downloader.py", line 114, in get_titles
    clips = extractor.extract(base_url, latest_only)
  File "/userpath/lib/python3.10/site-packages/yledl/extractors.py", line 248, in extract
    playlist = self.get_playlist(url, latest_only)
  File "/userpath/lib/python3.10/site-packages/yledl/extractors.py", line 252, in get_playlist
    return AreenaPlaylistParser(self.httpclient).get(url, latest_only)
  File "/userpath/lib/python3.10/site-packages/yledl/extractors.py", line 268, in get
    tree = self.httpclient.download_html_tree(url)
  File "/userpath/lib/python3.10/site-packages/yledl/http.py", line 71, in download_html_tree
    response = self.get(url, extra_headers)
  File "/userpath/lib/python3.10/site-packages/yledl/http.py", line 113, in get
    r.raise_for_status()
  File "/userpath/lib/python3.10/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://areena.yle.fi/1-1570236
```

---

Directly curl with the site got the same 403 response.
```
$ curl -I https://areena.yle.fi/1-1570236
HTTP/2 403
server: CloudFront
date: Tue, 01 Nov 2022 02:12:50 GMT
```

**Comment by aajanki** [comment]
This seems to be related to geo-blocking, because I get the 403 response only when running on a machine located outside Finland.

I made a change that mostly fixes this, but some tests seem to be flaky. Often they succeed but sometimes they fail with 403. I'll try to see if I can find a way to make them more reliably.

**Comment by aajanki** [comment]
I believe that this is now fixed in the latest version yle-dl 20221111.
